packages/firestore/src/local/indexeddb_persistence.ts

```diff
@@ -140,12 +142,15 @@ export class IndexedDbPersistence implements Persistence {
       .then(() => {
         this.scheduleOwnerLeaseRefreshes();
         this.attachWindowUnloadHook();
+      })
+      .then(() => {
+        this.hasStarted = true;
```

Comment:
This can be part of the previous `then` block.

But that being said, I don't this will work one I merge my multi-tab PR. In the new `scheduleOwnerLeaseRefreshes` I am using `started` to determine whether I should issue callbacks. We essentially need three states: not-started, starting, started. And since nothing else should be happening while we call `.start()`, it should be safe to keep flipping the started bit in the beginning?

Reply:
As discussed, leaving this at the end in its own `.then()` call.